src/main/java/com/ust/sourcecourse/configuration/service/DBDataSourceService.java

```diff
@@ -135,22 +135,37 @@ public ResponseEntity<String> removeTagFromSourceTable(Long uid, String tag) {
 
 	public List<DBTable> searchTablesByTag(String tag) {
 		List<SourceTable> sourcetables = sourceTableRepository.findAll();
-		return sourcetables.stream()
-				.filter(sourcetable -> sourcetable.getTags() != null && sourcetable.getTags().contains(tag))
-				.map(sourcetable -> getDBTable(sourcetable)).collect(Collectors.toList());
+		if (sourcetables.isEmpty()) {
+			throw new CustomException("No tables found in the database.");
+		}
+
+		List<DBTable> result = sourcetables.stream()
+				.filter(sourceTable -> sourceTable.getTags() != null && sourceTable.getTags().contains(tag))
+				.map(sourceTable -> getDBTable(sourceTable)).collect(Collectors.toList());
+		if (result.isEmpty()) {
+			throw new CustomException("No tables found with tag '" + tag + "'.");
+		}
+		return result;
 	}
 
 	public List<DBTableColumn> searchcolumnByTag(String tag) {
 		List<SourceColumn> sourcecolumns = sourceColumnRepository.findAll();
-		return sourcecolumns.stream()
+		if (sourcecolumns.isEmpty()) {
+			throw new CustomException("No tables found in the database.");
+		}
+		
+			List<DBTableColumn> result1 =sourcecolumns.stream()
 				.filter(sourcecolumn -> sourcecolumn.getTags() != null && sourcecolumn.getTags().contains(tag))
 				.map(sourcecolumn -> getDBTableColumn1(sourcecolumn)).collect(Collectors.toList());
-
+			if (result1.isEmpty()) {
+				throw new CustomException("No tables found with column tag '" + tag + "'.");
```

Comment:
No need of checking the condition again.

Reply:
resolved